### PR TITLE
Hardening collections

### PR DESCRIPTION
**Pull request by martinpovolny**
Test that primary collections have rbac id for GET read action.
    
Add a spec checking that each primary collection has an rbac identifier
for GET read action. So that data cannot be read w/o authorization.
    
There are exceptions to the rule: there's data accessible to any
authenticated user. Those collections are whitelisted in the test.

Also adding missing feature identifiers.

### CODE REVIEW

**Comment by abellotti** [comment]
❤️ it.  Thanks @martinpovolny for adding the missing identifiers and the spec.  

I think a couple of tests need the updated api_basic_authorize with the new roles + the rubocop above.

**Comment by martinpovolny** [comment]
* fixed the style issue
  * updated the specs that needed change
  * added specs covering the changes in authorization

File: spec/requests/roles_spec.rb [hunk 1]
```diff
@@ -244,6 +244,26 @@ def test_features_query(role, role_url, klass, attr = :id)
...
+      role = FactoryBot.create(:miq_user_role, :features => "miq_request_approval")
+
+      new_feature = {:identifier => "miq_request_view"}
+      url = api_role_features_url(nil, role)
+      post(url, :params => gen_request(:assign, new_feature))
+
+      expect(response).to have_http_status(:forbidden)
+
+      # Refresh the role object
+      role = MiqUserRole.find(role.id)
```
Comment (lpichler) [inline spec/requests/roles_spec.rb]: can be `role.reload` ? (then you can remove comment :) )

**Comment by simaishi** [comment]
Ivanchuk backport details:

```
$ git log -1
commit a27dbb8f115b1146d8d06e3b993a96d3cad9225b
Author: Libor Pichler <lpichler@redhat.com>
Date:   Fri Oct 25 15:06:05 2019 +0200

    Merge pull request #692 from martinpovolny/hardening_collections

    Hardening collections

    (cherry picked from commit 2d7f6ce87fb3f2f4b796d6e87d962114679b00d4)

    Fixes https://bugzilla.redhat.com/show_bug.cgi?id=1784102
```

**Pull request merged**